### DISCUSSION TITLE
[Question] No training effect using Github Actions?

### DISCUSSION THREAD

**Discussion by jonas4climate** [philosowaffle/peloton-to-garmin#464] [TE / TSS / Vo2 / Intensity Minutes]
**Problem**
I'm not sure if this is a user issue on my front but I set up Github Actions and imported my Peloton workouts. All data is displayed well and imported into my Garmin Connect but it doesn't appear to add Aerobic/Anaerobic Training Effect stats nor does it impact training readiness overall apparently. Is this currently not supported?

**Installation (please complete the following information):**
 - OS: N/A
 - Method: GitHub Action
 - Version: master
 - Watch: Fenix 7S

**Comment by jonas4climate** [comment]
Images of the stats it did and did not track:
<img width="1016" alt="Screenshot 2023-04-02 at 23 26 14" src="https://user-images.githubusercontent.com/44198728/229379945-dd30aca5-6c20-4cd7-a9ab-102968d4873c.png">
<img width="1015" alt="Screenshot 2023-04-02 at 23 28 26" src="https://user-images.githubusercontent.com/44198728/229379948-eb670dcd-1969-4003-a97e-6960ab077904.png">
<img width="1026" alt="Screenshot 2023-04-02 at 23 29 11" src="https://user-images.githubusercontent.com/44198728/229379950-5e32a88d-933e-4dc2-9cff-836eae3eb5d5.png">

**Comment by philosowaffle** [comment]
This is _sort of_ supported.  All calculations of TE/TSS/IF etc. are done by Garmin Connect itself, all P2G can do is provide as much data as possible (namely HR, Exercise Time, and Power Zone information) and hope that Garmin will do something with it.  Further reading on past investigations into this can be found [here](https://github.com/philosowaffle/peloton-to-garmin/issues/114) and [here](https://philosowaffle.github.io/peloton-to-garmin/faq.html#vo2-max-and-tss).  

I've never felt like I had a firm grasp on why this data works flawlessly for some and is missing for others.

For what little its worth, I am able to see Training Effect for my most recent Peloton cycling workouts (fenix 6 and 7).
![firefox_2023-04-07_09-03-43](https://user-images.githubusercontent.com/7275041/230622118-9ac63178-4c6b-4a8e-b160-1856f2f09ae9.png)

**Comment by gfedorko** [comment]
Have there been any recent discoveries on this? Previously, training effect worked fine for me using GitHub actions, but since Mid-October the training effect stopped working properly. I agree it is an issue on Garmin's side. I can actually see that the training effect is calculated on the Garmin app, but not on my watch (Forerunner 255). 

Could be tied to HR data - when I look at my workouts in the Health app or Zones app (iOS), the P2G activities do not have HR data associated with them, even though it is present in the Garmin app.

**Reply by seesomak** [reply]
Curious if there is any update from others having similar issues? Starting three days ago, training effect no longer shows up for my Peloton rides or activities through the Peloton App with HR data. It was working four days ago and I haven't changed any settings. The rides and activities are uploaded to Garmin Connect but just no training effect, which is odd.

**Reply by powerz20** [reply]
Same exact thing here. Have you figured it out yet?

**Reply by gfedorko** [reply]
@seesomak @powerz20  how are you syncing? Github Actions?

**Reply by seesomak** [reply]
@gfedorko Github actions. There's an issues thread opened on this but it seems like a Garmin issue. I posted a couple of links in the issues thread.

**Comment by iwoloschin** [comment]
I used my brother's Peloton Bike+ over Thanksgiving and got training effect status on my Forerunner 965. I do not get this on my Peloton Bike though. Looking at the generated FIT files (well, CSV conversion generated by FitCSVTool) I don't see any obvious differences. Happy to provide sample files or debug more if there's anywhere obvious to look.

**Reply by iwoloschin** [reply]
I'm running my own container if that makes any difference, though I don't expect it should.

**Reply by gfedorko** [reply]
Cant imagine that it would make a difference, but maybe I need to spend some time and set one up too rather than use GitHub actions.

**Reply by gfedorko** [reply]
[Here is a link to a related article ](https://support.garmin.com/en-US/?faq=apva7Th82K6Mm0zY4shzD6#:~:text=If%20the%20Training%20Status%20is,of%20your%20VO2%20max%20estimate.)

**Reply by iwoloschin** [reply]
Huh interestingly, I did a ride today (on my Bike) with no other activity and my Training Load has gone up. So it does work, sometimes? I have not tried pulling FIT files to investigate yet, can maybe do that later today, or probably over the weekend.

**Reply by gfedorko** [reply]
What method are you using to sync? Github Actions?

**Reply by iwoloschin** [reply]
I've been running in docker (on a VM) but I'm thinking of moving to Github Actions. I don't really believe this has an impact on what syncs though.

**Reply by philosowaffle** [reply]
Just to confirm for those questioning: how you run P2G (exe, docker, github action) will have no impact on this particular issue.  

The only functional difference between these three methods is the [GitHub action does not support MFA flow](https://philosowaffle.github.io/peloton-to-garmin/v4.0.0/install/).  Otherwise its the exact same code no matter how to run the application.

**Reply by geoffbon** [reply]
Are you using a custom deviceInfo.xml file?  Mine syncs all HR data from the Bike+ using the default device, and I get training effect data in the stats page on Garmin's connect website:
![image](https://github.com/philosowaffle/peloton-to-garmin/assets/54509333/095e710e-4fa0-4ca3-b7d1-0bed1c676627)

**Reply by jayflan** [reply]
Hi all.  

@geoffbon, I'm not seeing TE stats in my p2g rides and I'm using the default (Tacx) setup like you.    No deviceInfo.xml changes.  

I'll mention I'm using Docker, though I know it shouldn't matter.  Besides the TE, everything else comes across just fine.  

Seems that it works for some, but maybe not for others somehow.  I've been able to merge HR/TE from my Instinct 2X and a p2g file using gotoes.org to get TE in GC, but that's hopefully only a short-term workaround.

**Reply by jhealy2025** [reply]
@philosowaffle my last ride with Training Effect was December 8th.  Before and one that date every ride included TE.  After that, none do.  Did anything change around that date?

**Reply by philosowaffle** [reply]
For those still having trouble, are ya'll caught up on the workarounds and resolution to this issue over [here](https://github.com/philosowaffle/peloton-to-garmin/issues/578)?

**Reply by jhealy2025** [reply]
That worked. I apologize for not reading into it fully. When I saw a fix for TE and Load in the change log of the release version I assumed that was the same change. Thanks for your reply and getting this fixed.
